src/mdformat/renderer/__init__.py

```diff
@@ -100,8 +101,16 @@ def render_tree(
 
     @staticmethod
     def _write_references(env: MutableMapping) -> str:
+        def label_sort_key(label: str) -> str:
+            if not label:
+                raise AssertionError("link label cannot be empty string")
+            if all(c in digits for c in label):
```

Comment:
```suggestion
            if all(c in string.digits for c in label):
```
I prefer to keep the stdlib context here.